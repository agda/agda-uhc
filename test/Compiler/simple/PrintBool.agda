module PrintBool where

open import Common.IO
open import Common.Bool
open import Common.Char
open import Common.List
open import Common.Unit
open import Common.String

isNewline : Char -> Bool
isNewline '\n' = true
isNewline _    = false

sequence : {A : Set} -> List (IO A) -> IO (List A)
sequence [] = return []
sequence (x ∷ xs) =
    r <- x ,
    rs <- sequence xs ,
    return (r ∷ rs)

mapM : {A B : Set} -> (A -> IO B) -> List A -> IO (List B)
mapM f xs = sequence (map f xs)

printList : List Char -> IO Unit
printList xs =
  mapM printChar xs ,,
  printChar '\n'


main : IO Unit
main =
    printChar 'a' ,,
    printList ('a' ∷ 'b' ∷ 'c' ∷ []) ,,
    putStrLn "printBool" ,,
    printBool (isNewline '\n') ,,
    printBool (isNewline 'a') ,,
    return unit
