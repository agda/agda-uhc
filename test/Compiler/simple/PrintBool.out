EXECUTED_PROGRAM

ret > ExitSuccess
out > aabc
out > printBool
out > truefalse
